[DOCS] add releases page and modify README.md (#520)
apache/incubator-xtable: website/releases/release-0.1.0-incubating.mdx

---
title: "Release 0.1.0-incubating"
sidebar_position: 1
---

## [Release 0.1.0-incubating](https://github.com/apache/incubator-xtable/releases/tag/0.1.0-incubating) ([docs](https://xtable.apache.org/docs/how-to))

> This is the first official apache release for Apache XTable, an incubating project under the Apache Software Foundation.

Reviewer: `apache` should be capitalized? and do we need `(Incubating)`

> Apache XTable™ (Incubating) is a cross-table converter for table formats that facilitates omni-directional interoperability across data processing systems and query engines.

Reviewer: ```suggestion
Apache XTable™ (Incubating) facilitates omni-directional interoperability across data processing systems and query engines by allowing users to convert between open table formats without the need to rewrite any data files.
```

> Currently, Apache XTable™ supports widely adopted open-source table formats such as Apache Hudi, Apache Iceberg, and Delta Lake.

Reviewer: ```suggestion
Currently, Apache XTable™ (Incubating) supports the open-source table formats for Apache Hudi, Apache Iceberg, and Delta Lake.
```

## Features
Apache XTable™ (Incubating) provides users with the ability to translate metadata from one table format to another.

Apache XTable™ (Incubating) provides two sync modes, "incremental" and "full." The incremental mode is more lightweight and has better performance, especially on large tables. If there is anything that prevents the incremental mode from working properly, the tool will fall back to the full sync mode.

This sync provides users with the following:

1. Syncing of data files along with their column level statistics and partition metadata

> 2. Schema updates in the source are reflected in the target table metadata

Reviewer 2: @sagarlakshmipathy Should we link the features page here to avoid duplication ?

Reviewer 2: For the first release there's only features to list and the improvements we have made to the project.

Author: @vinishjail97 I think its better we have it in two places. My 2c, I'd think there would be people looking for features in one place, pointing them to release page would not be a good user exp. Then again, we can't add links to the features page from release notes as well.

I'm open to other ideas.

Reviewer 2: okay let's keep it like it this and not overthink too much now.

3. Metadata maintenance for the target table formats.
   * For Hudi, unreferenced files will be marked as [cleaned](https://hudi.apache.org/docs/hoodie_cleaner/) to control the size of the metadata table.
   * For Iceberg, snapshots will be [expired](https://iceberg.apache.org/docs/latest/maintenance/#expire-snapshots) after a configured amount of time.
   * For Delta, the transaction log will be [retained](https://docs.databricks.com/en/sql/language-manual/delta-vacuum.html) for a configured amount of time.

## Improvements
1. Added apache release guide and infra components to be compliant with ASF release process.

> 2. Fix bugs related to dependency convicts, few edge cases related when parsing column stats.

Reviewer: `convicts` is meant to be something else?

Reviewer 2: Yeah it was supposed to be `conflicts` :)

3. Improved README, docker demo and website docs based on feedback provided by users.
4. Refactored the codebase to follow apache naming practices.

> [apache-xtable-0.1.0-incubating.src (1).tgz](https://github.com/user-attachments/files/16709894/apache-xtable-0.1.0-incubating.src.1.tgz)

Reviewer: why is there a (1)?

Reviewer 2: This is a typo when copying the notes from GH release, will remove it.
